libsrc/common/atoi.s

```diff
@@ -54,9 +54,9 @@ L3:     iny
 L5:     stx     tmp1            ; remember sign flag
 
 L6:     lda     (ptr1),y        ; get next char
-        tax
-        lda     __ctype,x       ; get character classification
-        and     #$04            ; digit?
+                                ; get character classification
+        jsr     ctype_preprocessor_no_check
+        and     #CT_DIGIT       ; digit?
```

Comment:
This code uses "ctype" here, but bellow it assumes a digit is '0' to '9' (see the sbc #'0' at line 94), so it would be smaller (and faster) to simply " sbc #'0' / cmp #'9' "

Reply:
Not sure if I understand this.
You propose to change:
```
         jsr     ctype_preprocessor_no_check
        and     #CT_DIGIT       ; digit?
        beq     L8              ; done
```
to
```
        sbc     #'0'
        cmp     #'9'
```
I guess a 'clc' before 'sbc' is needed, so that a 'bcs L8' after the cmp would work?

Reply:
@dmsc is saying that `atoi()` didn't need to use the `isdigit()` function, at all!  There's a much simpler way to validate the input string:  get the next character, convert it to binary, then check the binary range.  If the value is 10 or greater, then the character wasn't a digit; `atoi()` has finished scanning the string.  If the value isn't 10 or greater, then save it in .X, multiply the running total by ten, add the saved value, and continue.

Reply:
Changed and 4 bytes saved.